### PR TITLE
Image popup fixes

### PR DESCRIPTION
**Pull request by tatiana-scandi**
Fixes issues: 
https://github.com/scandipwa/scandipwa/issues/3196
https://github.com/scandipwa/scandipwa/issues/3197
https://github.com/scandipwa/scandipwa/issues/3198
https://github.com/scandipwa/scandipwa/issues/3211

1. I made mousemove not trigger any visual effect if image is already fits by height and doesn't need to be scrolled.
2. Made image that has height less than window height visually centered. 
3. Made initial image position be calculated and recalculated on slide change, not only on mousemove
4. Made image position be recalculated on window size change.
5. Fixed error of image position be incorrect if image is after video (it turned out that locator for getting images list was incomlete)
6. Fixed issue of video not played im zoom. Previously popup closure occurred in this case, and it broke gallery display at all. What you need is just switch from zoomed mode to not zoomed.
7. Made top and bottom of zoomed image visible.

Also made zoomed popup fullscreen again + improved UX a bit.

### CODE REVIEW

**Comment by aleksandrakorolova** [comment]
1.Images disappear when clicking back 

**Steps:**
1. Open PDP
2. Open image on full-screen
3. Click on Back

**Actual result:**
Images disappear
https://drive.google.com/file/d/1oDa95zAscFdzHV1tntdUyc1BMFe7AtqM/view?usp=sharing

**Expected result**
User redirects to previous page

**Notes:**
User stays on PDP if he/she opens image -> closes it -> clicks back
_________________________________________
2.Slider with arrows is displayed if product has 6+ images. Appear unnecessary indent between the last image and arrow when clicking on arrow
**Steps:**
1. Open product with many images - https://pr-3238.d25kd40decjpue.amplifyapp.com/default/accessories/accessories-bags/travel-bags/travel-bag-1.html
2. Open image on full-screen

**Actual result:**
Slider with arrows is displayed. Appear unnecessary indent between the last image and arrow when clicking on arrow
https://drive.google.com/file/d/1GecC_2-9oRXT72j0IeyqH8P9kz784TPh/view?usp=sharing

**Expected result:**
Slider with arrows is displayed only if product has more than 17 images

**Comment by tatiana-scandi** [comment]
Issue 1 - fixed
Issue 2 - I rewrote carousel gallery additional logic so that it dynamically updates gallery additional icons on zoom and resize.

**Comment by aleksandrakorolova** [comment]
1.Images switch quickly when closing images pop-up on full-screen

**Steps:**
1. Open PDP
2. Open image on full-screen
3. Click on Back

**Actual result:**
Images switch quickly (Base image appears and switch to opened image)
https://drive.google.com/file/d/13M_s0Qaydlg9lHNze94Jmg-33eWOZi09/view?usp=sharing

**Expected result**
Opened image is displayed. Images don't switch
______________________________
2.User doesn't redirect to previous page when clicking back button if he/she opened image pop-up and closet it

**Steps:**
1.  Open PDP
2. Open image on full-screen
3. Close it (Click on Cross)
4. Click on back button

**Actual result:**
Nothing happens

**Expected result:**
User redirects to previous page

**Comment by tatiana-scandi** [comment]
3 - fixed
4 - fixed in a separate PR https://github.com/scandipwa/scandipwa/issues/3248

**Comment by aleksandrakorolova** [comment]
3. Still, sometimes can reproduce 
https://drive.google.com/file/d/1FUd1cbkZKk9jpf8sn9F3QjgcmRK2Lyks/view?usp=sharing

4. ok

5. Horizontal images are displayed at the top when slider is scrolling to selected image

**Steps:**

1. Open PDP - https://pr-3238.d25kd40decjpue.amplifyapp.com/default/accessories/accessories-bags/travel-bags/travel-bag-1.html
2. Open image on full-screen
3. Click on the last image
4. Click on the penultimate image
5. Click on the last image

**Actual result:**
Horizontal images are displayed at the top
https://drive.google.com/file/d/1wAYW3sPYyJWNdeMvuSboot2bll2IYeh1/view?usp=sharing

**Expected result:**
Slide on full-screen works the same as on PDP

**Comment by tatiana-scandi** [comment]
Issue 3: I increased delay for image appearance even more (so that all required calculations are made and translateX is applied: you need to wait for page to load first, then get size of the slider container, image width, image number, calculate translateX for setting image correct position, apply translateX - these actions are not immediate and take time). If after that blinking appears in very rare cases, I suggest to leave it as is, as even greater delay will make user look at nothing (white area) for too long => effect will be worse than from very rare not smooth animation.

Issue 5: Fixed

**Comment by Elena-Evsa** [comment]
Issue 3: ok
Issue 5: Fixed

**Issue 6:** 
Video is not launched if user click on 'Play' in zoomed mode
**Step to reproduce:** 
1. Open https://pr-3238.d25kd40decjpue.amplifyapp.com/default/travel-bag-1.html 
2. Click on any image to zoom it
3. Click on video in carousel 
4. Click Play
**Video:**
https://vimeo.com/603983790/c3137e826e
**Expected result**
Video is launched if user click on 'Play' in zoomed mode

**Comment by tatiana-scandi** [comment]
Issue 6: should work now.

**Comment by Elena-Evsa** [comment]
Issue 6: fixed

**Issue 7:** 
Now small part of second picture shown on slide with 1st picture if user zoom in/out page
Screenshot:
image
Steps to reproduce:
1. Open https://pr-3238.d25kd40decjpue.amplifyapp.com/default/travel-bag-1.html 
2. Click on wide image with not white background (as on screenshot)
3. Zoom in/out _page_ (in my case it is shown on 33%, 90%, 125%, 175%)

**Issue 8:**
Horizontal scroll appear is carousel which contains 21 pictures after some steps
Steps to reproduce:
1. Open https://pr-3238.d25kd40decjpue.amplifyapp.com/default/travel-bag-1.html 
2. Click on last image
3. Zoom in last image
Screenshot:
image

**Issue 9:**
Part of images situated after the borders of screen in carousel after some steps:
1. Open https://pr-3238.d25kd40decjpue.amplifyapp.com/default/travel-bag-1.html 
2. Click on last image
3. Zoom in last image
4. Go to the first image in carousel
5. Observe last image in carousel
Video: 
https://vimeo.com/604600531/db143245c9

**Issue 10:**
Left boarder of image in carousel is cut after steps:
1. Open https://pr-3238.d25kd40decjpue.amplifyapp.com/default/travel-bag-1.html 
2. Click on last image
3. Zoom in last image
4. To better observe, click on image with white background
Screenshot:
image

**Comment by Elena-Evsa** [comment]
Created separate bug report for issues 7-10 - #3328

File: packages/scandipwa/src/component/ProductGallery/ProductGallery.style.scss [hunk 1]
```diff
@@ -35,15 +35,30 @@
         @include mobile {
             display: none;
         }
+
+        &_isImageZoomPopupActive {
+            position: absolute;
+            bottom: 20px;
+            left: 20px;
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/ProductGallery/ProductGallery.style.scss:42]: Please use the logical css properties.

**Review by zans-laksa — CHANGES_REQUESTED** [review]
Did you also tested this on RTL layout?

**Comment by tatiana-scandi** [comment]
> Did you also tested this on RTL layout?

Yes (no new RTL related logic was added in this PR, the old one was not broken)

**Review by zans-laksa — APPROVED** [review]

**Pull request merged**